tests/test_insert_deletes_consolidate.cpp

```diff
@@ -79,59 +75,115 @@ inline void load_aligned_bin_part(const std::string& bin_file, T* data,
+              << "max points: " << report._max_points << std::endl
+              << "empty slots: " << report._empty_slots << std::endl
+              << "deletes processed: " << report._slots_released << std::endl
+              << "latest delete size: " << report._delete_set_size << std::endl
+              << "rate: (" << delete_points / report._time
+              << " points/second overall, "
+              << delete_points / report._time /
+                     delete_params.Get<unsigned>("num_threads")
+              << " per thread)" << std::endl;
+  } else if (insert_percentage > 0 && delete_percentage == 0) {
+    std::cout << " only insert randomly" << std::endl;
+#pragma omp parallel for num_threads(thread_count) schedule(dynamic)
+    for (int64_t i = build_points; i < insert_points + build_points; i++) {
+      index.insert_point(&data[i * aligned_dim], static_cast<TagT>(i));
+    }
+    const double elapsedSeconds = insert_timer.elapsed() / 1000000.0;
+    std::cout << "Insertion time " << elapsedSeconds << " seconds ("
+              << (max_points * insert_percentage) / elapsedSeconds
+              << " points/second overall, "
+              << (max_points * insert_percentage) / elapsedSeconds /
+                     thread_count
+              << " per thread)\n ";
   }
-  const double elapsedSeconds = insert_timer.elapsed() / 1000000.0;
-  std::cout << "Insertion time " << elapsedSeconds << " seconds ("
-            << (end - start) / elapsedSeconds << " points/second overall, "
-            << (end - start) / elapsedSeconds / thread_count
-            << " per thread)\n ";
 }
 
 template<typename T, typename TagT>
 void delete_from_beginning(diskann::Index<T, TagT>& index,
                            diskann::Parameters&     delete_params,
-                           size_t                   points_to_skip,
-                           size_t points_to_delete_from_beginning) {
+                           size_t max_points, float build_percentage,
+                           size_t points_to_skip, float delete_percentage) {
+  int64_t build_points = max_points * build_percentage;
   try {
     std::cout << std::endl
-              << "Lazy deleting points " << points_to_skip << " to "
-              << points_to_skip + points_to_delete_from_beginning << "... ";
-    for (size_t i = points_to_skip;
-         i < points_to_skip + points_to_delete_from_beginning; ++i)
-      index.lazy_delete(i);
-    std::cout << "done." << std::endl;
+              << "Lazy deleting points "
+              << " randomly ";
+    std::vector<int> delete_numbers;
+    for (int64_t j = 0; j < (int64_t) build_points - 1; j++) {
+      delete_numbers.push_back(j);
+    }
+    random_shuffle(delete_numbers.begin(), delete_numbers.end());
 
+    for (size_t i = points_to_skip;
+         i < (points_to_skip + max_points * delete_percentage); ++i) {
+      int64_t m = delete_numbers[i];
```

Comment:
this vector size is (build_points - 1), but the max index access here is (points_to_skip + max_points * delete_percentage), do you confirm it won't exceed the range?

Reply:
will not exceed the range because points_to_skip is 0 and delete_percentage<1